src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -390,11 +380,10 @@ static UCS_CLASS_INIT_FUNC(uct_cuda_ipc_iface_t, uct_md_h md, uct_worker_h worke
     self->config.max_poll     = config->max_poll;
     self->config.enable_cache = config->enable_cache;
 
+    self->map_memhandle   = uct_cuda_ipc_cache_map_memhandle;
     if (self->config.enable_cache) {
```

Comment:
I think we can remove function pointers. and use  uct_cuda_ipc_map_memhandle and uct_cuda_ipc_unmap_memhandle
directly. 
inside uct_cuda_ipc_unmap_memhandle() check  if (!region->refcount && !self->config.enable_cache) to close the handle.
may be later we can add other conditions ( #mem_handles(cache_size) ,#size_of_mapped_memory, etc) to keep check on amount of memory mapped.

WDYT?

Reply:
@bureddy do you see any harm in keeping the function pointers? I think it still provides flexibility (and potentially cleaner) if we decide to change something in the future.

Reply:
i'm ok with keeping the function pointers. how about removing  "self->config.enable_cache"  the condition here and move it  inside uct_cuda_ipc_unmap_memhandle .
here we can assign func pointer to generic names (uct_cuda_ipc_map_memhandle, uct_cuda_ipc_unmap_memhandle)